diff --git a/opal/mca/mpool/memkind/configure.m4 b/opal/mca/mpool/memkind/configure.m4
--- a/opal/mca/mpool/memkind/configure.m4
+++ b/opal/mca/mpool/memkind/configure.m4
@@ -30,7 +30,7 @@
-    AS_IF([test "$ompi_mpool_memkind_happy" = "yes"],
+    AS_IF([test "$opal_mpool_memkind_happy" = "yes"],
            [$1],
            [AS_IF([test -n "$with_memkind" && test "$with_memkind" != "no"],
-                 [AC_MSG_ERROR([Lustre support requested but not found.  Aborting])])
+                 [AC_MSG_ERROR([MEMKIND support requested but not found.  Aborting])])
             $2])
 
     # substitute in the things needed to build memkind
diff --git a/config/opal_check_libnl.m4 b/config/opal_check_libnl.m4
--- a/config/opal_check_libnl.m4
+++ b/config/opal_check_libnl.m4
@@ -45,3 +45,3 @@
-    AC_ARG_WITH([libn-libdirl],
+    AC_ARG_WITH([libnl-libdir],
                 [AS_HELP_STRING([--with-libnl-libdir=DIR],
                                 [Directory prefix for libnlv3 libs(typically only necessary if libnl is installed in a location that the compiler/linker will not search by default)])])
diff --git a/opal/mca/reachable/netlink/configure.m4 b/opal/mca/reachable/netlink/configure.m4
--- a/opal/mca/reachable/netlink/configure.m4
+++ b/opal/mca/reachable/netlink/configure.m4
@@ -29,3 +29,4 @@
-    AS_IF([test $opal_libnl_version -eq 3],
+    AS_IF([test $opal_libnl_version -eq 3 ||
+           { test $opal_libnl_version -ne 1 && test -n "$with_libnl" && test "$with_libnl" != "no"; }],
         [# Use the libnl base options for the route search
          with_libnl_route=${with_libnl}
diff --git a/configure b/configure
--- a/configure
+++ b/configure
@@ -2027,2 +2027,2 @@
-with_libn_libdirl
+with_libnl_libdir
 enable_silent_rules
@@ -5860,5 +5860,5 @@
-# Check whether --with-libn-libdirl was given.
-if test ${with_libn_libdirl+y}
+# Check whether --with-libnl-libdir was given.
+if test ${with_libnl_libdir+y}
 then :
-  withval=$with_libn_libdirl;
+  withval=$with_libnl_libdir;
 fi
@@ -105846,9 +105846,9 @@
-    if test "$ompi_mpool_memkind_happy" = "yes"
+    if test "$opal_mpool_memkind_happy" = "yes"
 then :
   should_build=$should_build
 else $as_nop
   if test -n "$with_memkind" && test "$with_memkind" != "no"
 then :
-  as_fn_error $? "Lustre support requested but not found.  Aborting" "$LINENO" 5
+  as_fn_error $? "MEMKIND support requested but not found.  Aborting" "$LINENO" 5
 fi
            should_build=0
@@ -112869,3 +112869,4 @@
-    if test $opal_libnl_version -eq 3
+    if test $opal_libnl_version -eq 3 ||
+           { test $opal_libnl_version -ne 1 && test -n "$with_libnl" && test "$with_libnl" != "no"; }
 then :
   # Use the libnl base options for the route search
diff --git a/3rd-party/prrte/src/mca/prtereachable/netlink/configure.m4 b/3rd-party/prrte/src/mca/prtereachable/netlink/configure.m4
--- a/3rd-party/prrte/src/mca/prtereachable/netlink/configure.m4
+++ b/3rd-party/prrte/src/mca/prtereachable/netlink/configure.m4
@@ -26,3 +26,7 @@
     prte_reachable_netlink_happy=1
 
+    with_libnl_route=${with_libnl}
+    with_libnl_route_libdir=${with_libnl_libdir}
+    with_libnl_route_incdir=${with_libnl_incdir}
+
     # ugly hack to deal with potentially alternate locations for
diff --git a/3rd-party/prrte/configure b/3rd-party/prrte/configure
--- a/3rd-party/prrte/configure
+++ b/3rd-party/prrte/configure
@@ -49483,3 +49483,7 @@
     prte_reachable_netlink_happy=1
 
+    with_libnl_route=${with_libnl}
+    with_libnl_route_libdir=${with_libnl_libdir}
+    with_libnl_route_incdir=${with_libnl_incdir}
+
     # ugly hack to deal with potentially alternate locations for
