diff --git a/meson.build b/meson.build
index a0cf675..ba359aa 100644
--- a/meson.build
+++ b/meson.build
@@ -226,11 +226,6 @@ foreach tuple : _customdirs
   set_variable(dir_option, customdir)
 endforeach
 
-foreach confdir : [sysconfdir, distconfdir]
-  install_emptydir(confdir / 'depmod.d')
-  install_emptydir(confdir / 'modprobe.d')
-endforeach
-
 _completiondirs = [
   ['bash', 'bash-completion', 'bash-completion' / 'completions', '@0@'],
   ['fish', 'fish',            'fish' / 'vendor_functions.d',     '@0@.fish'],
@@ -421,7 +416,7 @@ endif
 
 install_headers('libkmod/libkmod.h')
 
-libkmod = shared_library(
+libkmod = library(
   'kmod',
   libkmod_files,
   dependencies : libkmod_deps + cdeps,
@@ -492,13 +487,15 @@ if module_compressions != ''
   _kmod_unescaped_variables += ['module_compressions=' + module_compressions]
 endif
 
-pkg.generate(
-  name : 'kmod',
-  description : 'Tools to deal with kernel modules',
-  install_dir : datadir / 'pkgconfig',
-  unescaped_variables : _kmod_unescaped_variables,
-  variables : _kmod_variables,
-)
+if get_option('tools')
+  pkg.generate(
+    name : 'kmod',
+    description : 'Tools to deal with kernel modules',
+    install_dir : datadir / 'pkgconfig',
+    unescaped_variables : _kmod_unescaped_variables,
+    variables : _kmod_variables,
+  )
+endif
 
 _tools = [
   'depmod',
@@ -527,7 +524,7 @@ foreach tool : _tools
     command : ['ln', '-sf', kmod, '@OUTPUT@'],
     output : tool,
     depends : kmod,
-    build_by_default : true,
+    build_by_default : get_option('tools'),
   )
 endforeach
 
